api/room_resource/schema.py

```diff
@@ -79,6 +79,48 @@ def mutate(self, info, **kwargs):
         return AssignResource(room_resource=room_resources)
 
 
+class UpdateAssignedResource(graphene.Mutation):
+    """
+        Update assigned resource in a room
+    """
+    class Arguments:
+        resource_id = graphene.Int(required=True)
+        room_id = graphene.Int(required=True)
+        quantity = graphene.Int(required=True)
+    room_resource = graphene.Field(RoomResource)
+
+    @Auth.user_roles('Admin')
+    def mutate(self, info, room_id, resource_id, **kwargs):
+        exact_resource = ResourceModel.query.filter_by(
+            id=resource_id).first()
+        room_resource_query = RoomResource.get_query(info)
+        rooms_assigned_to_resource = room_resource_query.filter(
+            RoomResourceModel.room_id == room_id).all()
+        if not rooms_assigned_to_resource:
+            raise GraphQLError('Room has no assigned resource')
+        current_resource = None
+        for room_resource in rooms_assigned_to_resource:
+            if room_resource.resource_id == resource_id:
+                current_resource = room_resource.quantity
+                break
```

Comment:
@kwanj-k Kindly rename the variable `rooms_assigned_to_resource` as we can only assign resources to rooms and not the other way. You can use `rooms_with_resource`

Reply:
Done